src/conn/conn_reconfig.c

```diff
@@ -94,64 +90,61 @@ __wt_conn_compat_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfi
      */
     WT_RET(__wt_config_gets(session, cfg, "compatibility.require_min", &cval));
     if (cval.len != 0)
-        WT_RET(__conn_compat_parse(session, &cval, &min_major, &min_minor));
+        WT_RET(__conn_compat_parse(session, &cval, &min_compat.major, &min_compat.minor));
 
     WT_RET(__wt_config_gets(session, cfg, "compatibility.require_max", &cval));
     if (cval.len != 0)
-        WT_RET(__conn_compat_parse(session, &cval, &max_major, &max_minor));
+        WT_RET(__conn_compat_parse(session, &cval, &max_compat.major, &max_compat.minor));
 
     /*
      * The maximum required must be greater than or equal to the compatibility release we're using
      * now. This is on an open and we're checking the two against each other. We'll check against
      * what was saved on a restart later.
      */
```

Comment:
The following four checks are repeated logic and can be simplified, but doing so up front makes the diffs hard to read. Once everyone's had a chance to look over this section I'll push an additional commit that can be checked independently.

Reply:
On second thought this PR is pretty large as is. I'll move this change to a second ticket.